### ISSUE TITLE
Cancellation doesn't survive its scope

### ISSUE THREAD

**Issue by smurfix** [agronholm/anyio#24]
This should work, but doesn't. I have no good idea why because AFAIU cancelling an asyncio task is a one-stop thing which raises a `CancelledError` and *clears* the task's `must_cancel` flag.
```
import asyncio
import anyio

async def main():
    try:
        async with anyio.fail_after(0.1):
            await anyio.sleep(1)
    except TimeoutError:
        print("Timed out.")
    await anyio.sleep(0.2)
    print("Coninued OK.")

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

**Comment by Fuyukai** [comment]
Not sure if this is known already, but this also happens on curio.

**Comment by smurfix** [comment]
should be fixed by PR#25, at least on asyncio
